ArdusatSDK.h

```diff
@@ -49,152 +66,532 @@ typedef enum {
+    const char * toJSON(const char * sensorName);
+};
+
+
+/**************************************************************************//**
+ * @class Pressure
+ * @ingroup sensor
+ *
+ * @defgroup pressure
+ * @brief Encapsulates all functionality related to the Pressure Sensor
+ *
+ * This class can be used to initialize, further configure, read, and print
+ * data receieved from the BMP180 on the Adafruit 9 Degrees of Freedom
+ * Board.
+ *
+ * @note This Sensor is not available on the SpaceBoard
+ *
+ * Example Usage:
+ * @code
+ *     Pressure press;                               // Instantiate sensor object
+ *     press.begin();                                // Initialize sensor
+ *     Serial.println(press.readToJSON("pressure")); // Read and print values in JSON
+ * @endcode
+ *****************************************************************************/
+class Pressure: public Sensor {
+  protected:
+    bmp085_mode_t bmp085_mode;
+
+    boolean initialize(void);
+    boolean readSensor(void);
+
+  public:
+    float pressure;
+    Pressure(void);
+    Pressure(bmp085_mode_t mode);
+
+    float altitudeFromSeaLevelPressure(float seaLevelPressure);
+    float seaLevelPressureFromAltitude(float altitude);
+
+    const char * toCSV(const char * sensorName);
+    const char * toJSON(const char * sensorName);
+};
+
+
+/**************************************************************************//**
+ * @class RGBLight
+ * @ingroup sensor
+ *
+ * @defgroup rgblight
+ * @brief Encapsulates all functionality related to the RGBLight Sensor
+ *
+ * This class can be used to initialize, further configure, read, and print
+ * data receieved from the either the TCS34725 board (default) or the ISL29125
+ * board
+ *
+ * @note No RGBLight Sensors are provided with the Space Kit
+ *
+ * Example Usage:
+ * @code
+ *     RGBLight rgb;                          // Instantiate sensor object
+ *     rgb.begin();                           // Initialize sensor
+ *     Serial.println(rgb.readToJSON("rgb")); // Read and print values in JSON
+ * @endcode
+ * Or
+ * @code
+ *     RGBLightTCS rgb;                       // Instantiate sensor object
+ *     rgb.begin();                           // Initialize sensor
+ *     Serial.println(rgb.readToJSON("rgb")); // Read and print values in JSON
+ * @endcode
+ * Or
+ * @code
+ *     RGBLightISL rgb;                       // Instantiate sensor object
+ *     rgb.begin();                           // Initialize sensor
+ *     Serial.println(rgb.readToJSON("rgb")); // Read and print values in JSON
+ * @endcode
+ *****************************************************************************/
+class RGBLight: public Sensor {
+  protected:
+    uint8_t islIntensity;
+    tcs34725IntegrationTime_t tcsIt;
+    tcs34725Gain_t tcsGain;
+
+    boolean initialize(void);
+    boolean readSensor(void);
+    RGBLight(sensor_id_t sensor_id, uint8_t islIntensity, tcs34725IntegrationTime_t tcsIt, tcs34725Gain_t tcsGain);
```

Comment:
Let's leave this constructor out altogether, I think it's confusing to have parameters for both sensors here. I think the appropriate paradigm is for the sensor-specific derived classes to handle all config options, if you are using this generic class, you don't manually configure the sensor.